alfresco-transform-misc/alfresco-transform-misc/src/main/java/org/alfresco/transformer/transformers/TextToPdfContentTransformer.java

```diff
@@ -146,7 +151,129 @@ protected InputStreamReader buildReader(InputStream is, String encoding)
             }
             if (charset != null)
             {
-                logger.debug("Processing plain text in encoding " + charset.displayName());
+                // Handles the situation where there is a BOM even though the encoding indicates that normally
+                // there should not be one for UTF-16BE and UTF-16LE. For extra flexibility includes UTF-16 too
+                // which optionally has the BOM. Rather than look at the BOM we look at the number of zero bytes
+                // in the first few character. XML files even when not in European languages tend to have more
+                // even zero bytes when big-endian encoded and more odd zero bytes when little-endian.
+                // Think of: <?xml version="1.0"?> The normal Java decoder does not have this flexibility but
+                // other transformers do.
+                String name = charset.displayName();
+                if ("UTF-16".equals(name) || "UTF-16BE".equals(name) || "UTF-16LE".equals(name))
+                {
+                    logger.debug("Handle big and little endian UTF-16 text. Using UTF-16 rather than encoding " + name);
+                    charset = Charset.forName("UTF-16");
+                    is = new PushbackInputStream(is, UTF16_READ_AHEAD_BYTES)
+                    {

```

Comment:
Added a PushbackInputStream to change the stream of bytes on the fly so that the Reader would have no  problem with all the commonly found deviations from the standards. Java follows the standard to the letter unlike other transformers.